chain/network/src/peer_manager/network_state/mod.rs

```diff
@@ -558,11 +565,29 @@ impl NetworkState {
     /// Return whether the message is sent or not.
     /// The message might be sent over TIER1 or TIER2 connection depending on the message type.
     pub fn send_message_to_account(
-        &self,
+        self: &Arc<Self>,
         clock: &time::Clock,
         account_id: &AccountId,
         msg: RoutedMessageBody,
     ) -> bool {
+        // If the message is allowed to be sent to self, we handle it directly.
+        if self.config.validator.account_id().is_some_and(|id| &id == account_id) {
+            // For now, we don't allow all types of messages to be sent to self.
+            debug_assert!(msg.allow_sending_to_self());
+            let this = self.clone();
+            let clock = clock.clone();
+            actix::spawn(async move {
+                this.receive_routed_message(
+                    &clock,
+                    this.config.node_id(),
+                    CryptoHash::default(),
```

Comment:
I'm not sure what should we put here for the msg_hash?

Reply:
See here:

https://github.com/near/nearcore/blob/c90b9ad06cdc7d202ca79ab6817351af4fcdd5b2/chain/network/src/peer/peer_actor.rs#L1108